zero/src/prover_state/mod.rs

```diff
@@ -31,6 +31,9 @@ use crate::prover_state::persistence::{
     VerifierResource,
 };
 
+// Default degree of the recursive circuit when a proof is missing from AllProof
+const DEFAULT_CIRCUIT_DEGREE: usize = 8;
```

Comment:
Why would we want to load anything if a proof is missing? Also this may crash unexpectedly in `load_table_circuits` if we specify a config with a range (for Keccak for instance) that excludes this value.

Reply:
We still need a recursive circuit to generate a dummy proof for the root circuit. Let me refactor the code to optimize this while also addressing the other comment.